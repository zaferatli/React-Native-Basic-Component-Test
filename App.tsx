 import React, {useState} from 'react';
import type {PropsWithChildren} from 'react';
import {Text, SafeAreaView, StyleSheet, View, TextInput} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';
import Button from './app/component/button';

type SectionProps = PropsWithChildren<{
  title: string;
}>;

function App(): JSX.Element {
  const [displayText, setDisplayText] = useState<String>();
  const [inputText, setInputText] = useState<String>();

  return (
    <SafeAreaView
      style={{
        backgroundColor: Colors.white,
      }}>
      <Text style={styles.textHeader}>React Native Component Testing</Text>
      <View style={styles.sectionContainer}>
        <TextInput
          testID={'TextInputTestID'}
          onChangeText={text => setInputText(text)}
          style={styles.textInput}
        />
        <Button
          text="Yazılanı Aktar"
          actionClick={() => setDisplayText(inputText)}
        />
        <Text testID={'DisplayTextTestID'} style={styles.text}>
          {displayText}
        </Text>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: '15%',
    width: '90%',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
  },
  textInput: {
    borderWidth: 1,
    height: 45,
    width: '100%',
    marginBottom: '5%',
    borderRadius: 10,
    borderColor: 'gray',
    textAlign: 'center',
  },
  text: {
    marginTop: '5%',
  },
  textHeader: {
    alignSelf: 'center',
  },
});

export default App;
