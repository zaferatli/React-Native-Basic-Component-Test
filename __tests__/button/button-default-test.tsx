import 'react-native';
import React from 'react';
import App from '../../App';
import Button from '../../app/component/button';

import { fireEvent, render } from '@testing-library/react-native';

it('renders correctly', () => {
    const { getByTestId, getByText, getAllByText } = render(<App />);

    let exampleText = 'Ã–rnek text';

    const testInputTestElementi = getByTestId('TextInputTestID');
    fireEvent.changeText(testInputTestElementi, exampleText)

    const testButtonTestElementi = getByTestId('ForTestCustomButtonID');
    fireEvent.press(testButtonTestElementi);

    const controlResult = getAllByText(exampleText);
    expect(controlResult).toHaveLength(1)

    expect(getByText(exampleText)).toBeTruthy()

});


it('button snapshot test for text ve click action', () => {
    const buttonShadowTest = render(<Button text={'Shadow test texti'} actionClick={() => console.log('ok')} />);

    expect(buttonShadowTest.toJSON()).toMatchSnapshot()
})

it('button shadow test for second text', () => {
    const buttonShadowTest = render(<Button text={'Shadow test texti'} actionClick={() => console.log('ok')} secondText={true} />);

    expect(buttonShadowTest.toJSON()).toMatchSnapshot()
})