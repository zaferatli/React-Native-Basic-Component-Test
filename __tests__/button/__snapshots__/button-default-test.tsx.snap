// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button shadow test for second text 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "tomato",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "width": "90%",
    }
  }
  testID="ForTestCustomButtonID"
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    Shadow test texti
  </Text>
  <Text>
    Ä°kinci text
  </Text>
</View>
`;

exports[`button snapshot test for text ve click action 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "tomato",
      "borderRadius": 10,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "width": "90%",
    }
  }
  testID="ForTestCustomButtonID"
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    Shadow test texti
  </Text>
</View>
`;
